tests/source/brsHamcrest/brsHamcrest_Matchers/Test_brsHamcrest_ObjectMatchers.brs

```diff
@@ -60,6 +79,25 @@ sub test_sameObjectAs_differentInstances (t as Object)
 end sub
 
 
+sub test_sameObjectAs_differentArrayInstances (t as Object)
+    test = setup_brsHamcrest_ObjectMatchers()
+
+    'GIVEN'
+    testTarget = [{foo: "bar"}, 2, true]
+    testValue = [{foo: "bar"}, 2, true]
+
+    'WHEN'
+    result = sameObjectAs(testValue).doMatch(testTarget)
+
+    'THEN'
+    t.assertFalse(result)
```

Comment:
Shouldn't the result be true? What's the difference between the two arrays?

Reply:
The difference is that they are not the same arrays, just identical arrays. Even if they share elements that are actually references to the same objects. 

But I guess it is important to note that the comparison is specifically the top level array in this case. If this was two different but identical arrays with 3 items, where all 6 items were a reference to the same other-array, then they would still be two different arrays and the match would fail.

Reply:
Again, you're correct, I misread the test!